### PR TITLE
Image - fix flaky tests

### PR DESCRIPTION
**Pull request by nitayneeman**
# ✨ Pull Request

### 💁 Description

This PR should fix the flaky tests in Image.

Those are the problematic tests:
- Image/Relative Media URI/with wiring: override background color
- Image/Absolute URL/with wiring: override background color
- Image/Relative Media URI: with onError
- Image/Relative Media URI: with onError

The common for all of them is the usage of `invalidSrc` which triggers the native broken image icon of the browser. The thing is ui-core's Image [replaces](https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/components/image/image.tsx#L113-L116) the source in such cases with an image with dimensions of 1x1 pixels, in order to hide the native icon. Probably sometimes Applitools is fast enough to take the screenshot before that the icon is replaced (I also managed to notice it when running the visuals locally).

I believe the 500ms with `await` should fix the problem because `onError` is invoked in ui-core's Image after setting the state (inside ui-core's Image). In case of the "override background color" I changed it to a non-broken source and simply hide the image to show the wired background.

### 👀  PR is ready for review 

- [X] Ready for review

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_656/

**Comment by nitayneeman** [comment]
Both visuals of "override background color" should break because now we're not using the `invalidSrc` so we don't have the red border anymore.

File: src/components/Image/Image.visual.tsx [hunk 1]
```diff
@@ -29,8 +30,11 @@ type ImageWithWrapperProps = ImageProps & {
 class ImageWithWrapper extends React.Component<ImageWithWrapperProps> {
   state = { hasError: false };
 
-  onError(onError) {
-    this.setState({ hasError: true }, () => setTimeout(() => onError(), 500));
+  async onError(onError) {
+    this.setState({ hasError: true }, async () => {
+      await delay(500);
+      onError();
+    });
```
Comment (jonathanadler) [inline src/components/Image/Image.visual.tsx:37]: Isn't it problematic that `onError` is defined twice?
```suggestion
  async onError(callback) {
    this.setState({ hasError: true }, async () => {
      await delay(500);
      callback();
    });
```

File: src/components/Image/Image.visual.tsx [hunk 2]
```diff
@@ -164,7 +168,7 @@ visualize('Image', () => {
       story('with wiring', () => {
         snap('override background color', (done) => (
           <ImageWithWrapper
-            src={invalidSrc}
+            src={src}
```
Comment (jonathanadler) [inline src/components/Image/Image.visual.tsx:171]: oh I thought the invalidSrc was to test the error state, no?

Comment (nitayneeman) [reply src/components/Image/Image.visual.tsx:171]: Yep it is and it's still being used for the error state. But I also used it to test the background color wiring.

File: src/components/Image/Image.visual.tsx (hunk 1 above)
Comment (nitayneeman) [reply src/components/Image/Image.visual.tsx]: Oops! I did the fix in a different PR instead 🤦🏻‍♂️:
https://github.com/wix/wix-ui-tpa/pull/651/files#diff-59cb9bbb8734f702d7df647b858b4173e292160bcc9615bbb390a3eecd5a7aabR32

I just pushed a fix to this branch.

**Review by jonathanadler — APPROVED** [review]

**Pull request merged**